src/vs/editor/contrib/suggest/suggestWidget.ts

```diff
@@ -162,14 +168,15 @@ export class SuggestWidget implements IDisposable {
 			persistHeight = persistHeight || !!e.north || !!e.south;
 			persistWidth = persistWidth || !!e.east || !!e.west;
 			if (e.done) {
-
-				// only store width or height value that have changed
+				// only store width or height value that have changed and also
+				// only store changes that are above a certain threshold
+				const threshold = Math.floor(this.getLayoutInfo().itemHeight / 3);
```

Comment:
This threshold might be too aggresive for high suggets widget.
E.G my widget is 600 px, I reduce the size to 400 px and the size is not stored.
I suggest to go with item size / 4 or /5

Also should the width threshold be a function of the width?

Reply:
The threshold is based on the item height, that's the height of a single row, not the height of the widget. With defaults it should be 6px but it will scale when larger/smaller fonts are used.

Reply:
Ok, makes sense, but then why not take the height of the whole item? Currently you just take 1/3 of it?

Reply:
I think the full height is too much, I was hovering between 1/3 and 1/2 and ended with 1/3 since that's already something, e.g. half the icon size